Unix/tests/wsman/test_wsmanparser.cpp

```diff
@@ -1487,6 +1487,138 @@ NitsTestWithSetup(TestFaultResponse2, TestParserSetup)
+        PAL_T("<p:ProviderName xsi:type=\"cim:cimString\">CIMWin32</p:ProviderName><p:StatusCode xsi:nil=\"true\" xsi:type=\"cim:cimUnsignedInt\"/>")
+        PAL_T("</p:__ExtendedStatus></f:ExtendedError></f:ProviderFault>")
+        PAL_T("</f:Message></f:WSManFault></s:Detail></s:Fault></s:Body></s:Envelope>");
+
+    memset(&wsheaders, 0, sizeof(wsheaders));
+
+    XML_Init(&xml);
+
+    XML_RegisterNameSpace(&xml, 's',
+                          ZT("http://www.w3.org/2003/05/soap-envelope"));
+    
+    XML_RegisterNameSpace(&xml, 'a',
+                          ZT("http://schemas.xmlsoap.org/ws/2004/08/addressing"));
+
+    XML_RegisterNameSpace(&xml, 'w',
+                          ZT("http://schemas.xmlsoap.org/ws/2005/06/management"));
+
+    XML_RegisterNameSpace(&xml, 'b',
+                          ZT("http://schemas.dmtf.org/wbem/wsman/1/cimbinding.xsd"));
+
+    XML_SetText(&xml, data);
+
+
+    if (!NitsCompare(MI_RESULT_OK, WS_ParseSoapEnvelope(&xml), PAL_T("Parse soap envelope error")))
+    {
+        goto cleanup;
+    }
+
+    if (!NitsCompare(MI_RESULT_OK, WS_ParseWSHeader(&xml, &wsheaders, USERAGENT_UNKNOWN), PAL_T("Parse header error")))
+    {
+        goto cleanup;
+    }
+
+    if (!NitsCompare(MI_TRUE, wsheaders.foundAction, PAL_T("No action in header")))
+    {
+        goto cleanup;
+    }
+    if (!NitsCompare(wsheaders.rqtAction, WSMANTAG_ACTION_FAULT_ADDRESSING, PAL_T("Action is not Fault ADDRESSING Response")))
+    {
+        goto cleanup;
+    }
+
+    if (!NitsCompare(MI_RESULT_OK, WS_ParseFaultBody(&xml, &fault), PAL_T("Unable to retrieve fault")))
+    {
+        goto cleanup;
+    }
+    if (!NitsCompareString(fault.code, PAL_T("http://www.w3.org/2003/05/soap-envelope:Sender"), 
+                           PAL_T("Mismatch of fault code")))
+    {
+        goto cleanup;
+    }
+    if (!NitsCompareString(fault.subcode, 
+                           PAL_T("http://schemas.xmlsoap.org/ws/2004/08/addressing:DestinationUnreachable"), 
+                           PAL_T("Mismatch of fault subcode")))
+    {
+        goto cleanup;
+    }
+    if (!NitsCompareString(fault.reason, PAL_T("The WS-Management service cannot process the request. The service cannot find the resource identified by the resource URI and selectors. "), PAL_T("Mismatch of fault reason")))
+    {
+        goto cleanup;
+    }
+    if (!NitsCompareString(fault.detail, PAL_T("CIMWin32"), 
+                           PAL_T("Invalid fault detail")))
+    {
+        goto cleanup;
+    }
+    if (!NitsCompare(MI_RESULT_OK, GetWsmanErrorFromSoapFault(fault.code, fault.subcode, fault.detail, &errorType, &errorTypeStr),
+                     PAL_T("Cannot determine errorCode")))
+    {
+        goto cleanup;
+    }
+    if (!NitsCompare(ERROR_WSMAN_DESTINATION_UNREACHABLE, errorType, PAL_T("Wrong error type")))
+    {
+        goto cleanup;
+    }
+
+    if (!NitsCompare(0, fault.mi_result, PAL_T("Wrong result code")))
```

Comment:
In fact I see this is the same as the previous unittest, and that does not seem correct either. I think there may have been a bug in the code from the previous change that needs fixing